presto-native-execution/presto_cpp/main/types/PrestoToVeloxPlanValidator.cpp

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/types/PrestoToVeloxPlanValidator.h"
+
+namespace facebook::presto {
+bool PrestoToVeloxPlanValidator::validatePlanFragment(
+    const velox::core::PlanFragment& fragment) {
+  return true;
```

Comment:
If there are multiple validations like 1. Don't allow TIMESTAMPTZ 2. Don't allow NLJ 3. Allow only particular combination of 1 and 2, then there would be a config for each validation and all checks in the same function. Did you consider allowing multiple validators instead ?

Also, given that I think this class would need to be aware of SystemConfig.